src/machines/workflow_machines.rs

```diff
@@ -516,12 +488,34 @@ impl WorkflowMachines {
 
             if event.event_type == EventType::WorkflowTaskStarted as i32 && next_event.is_none() {
                 self.handle_event(event, false)?;
-                return Ok(());
+                break;
             }
 
             self.handle_event(event, next_event.is_some())?;
         }
 
+        // Scan through to the next WFT, searching for any change markers, so that we can
+        // pre-resolve them.
+        for e in self.last_history_from_server.peek_next_wft_sequence() {
+            if let Some((change_id, deprecated)) = e.get_changed_marker_details() {
+                self.encountered_change_markers.insert(
+                    change_id.clone(),
+                    ChangeInfo {
+                        deprecated,
+                        created_command: false,
+                    },
+                );
+                // Found a change marker
+                self.drive_me
+                    .send_job(wf_activation_job::Variant::ResolveHasChange(
```

Comment:
Is it guaranteed that this job won't be received by lang too early? Or are we expecting lang to cache change statuses?

Reply:
It needs to cache them